sahi/slicing.py

```diff
@@ -519,141 +519,94 @@ def cal_aspect_ratio_orientation(width: int, height: int) -> str:
         image capture orientation
     """
 
-    def gcd(a, b):
-        """The GCD (greatest common divisor) is the highest number that evenly divides both width and height."""
-        return a if b == 0 else gcd(b, a % b)
-
-    r = gcd(width, height)
-    x = int(width / r)
-    y = int(height / r)
-    if x < y:
+    if width < height:
         return "vertical"
-    elif x > y:
+    elif width > height:
         return "horizontal"
     else:
         return "square"
 
 
-def low_resolution(height: int, width: int):
+def calc_slice_and_overlap_params(resolution: str, height: int, width: int, orientation: str) -> List:
     """
-
+    This function calculate according to image resolution slice and overlap params.
     Args:
-        height:
-        width:
+        resolution: str
+        height: int
+        width: int
+        orientation: str
 
     Returns:
-        default overlap params
+        x_overlap, y_overlap, slice_width, slice_height
     """
-    slice_height = height // 1
-    slice_width = width // 1
 
-    x_overlap = slice_width * 1
-    y_overlap = slice_height * 1
+    if resolution == "low":
```

Comment:
Thanks for the update. One slight improvement could be to define `split_row`, `split_col`, `overlap_width_ratio`, `overlap_height_ratio` under if statements, and then after if blocks, perform the following only once, no need to repeat this under each if block.

```python
        slice_height = height // split_col  # noqa
        slice_width = width // split_row  # noqa

        x_overlap = int(slice_width * overlap_width_ratio)  # noqa
        y_overlap = int(slice_height * overlap_height_ratio)  # noqa
```